<a href="https://colab.research.google.com/github/olegtyulyaev/--2/blob/main/Probability_theory_HW1%2C2%2C3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Теория вероятностей**



## Домашнее задание №1

**Задача 1.**

 Из колоды в 52 карты извлекаются случайным образом 4 карты.

1.   Найти вероятность того, что все 4 карты – крести
2.   Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

In [ ]:
from math import factorial
a = factorial(13) // (factorial (4) * factorial (13-4) )
b = factorial(39) // (factorial (0) * factorial (39-0) )

c = factorial(52) // (factorial (4) * factorial (52-4) )
(a*b)/c

0.0026410564225690276

In [ ]:
a = factorial(48) // (factorial (4) * factorial (48-4))
b = factorial(52) // (factorial (4) * factorial (52-4))
1-(a/b)

0.2812632745405854

**Ответ**
1.   P (4 трефы) = 0,26%
2.   P (хотя бы 1 туз) = 28,13%


**Задача 2.**

На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно . Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

In [ ]:
a = factorial(10) / (factorial (3) * factorial (10-3))
a

120.0

**Ответ**

120 возможных исходов, из которых один благоприятный, искомая вероятность равна 1/120 или 0,83%

**Задача 3.**

В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все 3 извлеченные детали окрашены?

In [ ]:
a = factorial(9) // (factorial (3) * factorial (9-3))
b = factorial(6) // (factorial (0) * factorial (6-0) )

c = factorial(15) // (factorial (3) * factorial (15-3))

(a*b)/c

0.18461538461538463

**Ответ**

P (3 из 3х окрашены) = 18,46%

**Задача 4.**

В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

In [ ]:
a = factorial(2) // (factorial (2) * factorial (2-2) )
b = factorial(98) // (factorial (0) * factorial (98-0) )

c = factorial(100) // (factorial (2) * factorial (100-2) )

(a*b)/c

0.00020202020202020202

**Ответ**

P (оба билеты выигрышные) = 0,02%


## Домашнее задание №2

**Задача 2.1**

 Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

In [ ]:
from math import factorial, exp
def bernulli(n, k, p):
    result =factorial(n)/(factorial(k)*factorial(n-k))
    return result*(p**k)*(1-p)**(n-k)

# Формула биномиального распределения: P(n)(X=k) = С(n,k) * p^k * q^(n-k) k = 85; n = 100; p = 0.8; q = 1-0.8 = 0.2

In [ ]:
print(f' P ={bernulli(100,85,0.8): .5f}')

 P = 0.04806


**Задача 2.2**

Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?

In [ ]:
p = 0.0004
n = 5000
lamda = p * n
lamda

#формула Пуассона  P(m) = (lambda^m / m!) * e^(-lambda)

2.0

In [ ]:
import math
P_0 = (lamda**0/(factorial(0))) * (math.e **-lamda)
print(f' P ={(P_0): .5f}')


 P = 0.13534


In [ ]:
P_2 = (lamda**2/(factorial(2))) * (math.e **-lamda)
print(f' P ={(P_2): .5f}')


 P = 0.27067


**Ответ**

P (0 лампочек перегорит) = 13,5%

P (2 лампочки перегорят) = 27,1%

**Задача 2.3**

Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

In [ ]:
print(f'P = {bernulli(144,70,0.5): .5f}')

P =  0.06281


**Ответ**

Вероятность, что орел выпадет ровно 70 раз = 6,28%

**Задача 2.4**

В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча.

* 2.4.1 Какова вероятность того, что все мячи белые?
* 2.4.2 Какова вероятность того, что ровно два мяча белые?
* 2.4.3 Какова вероятность того, что хотя бы один мяч белый?

In [ ]:
p4=7/10*6/9*9/11*8/10
p2=7/10*6/9*2/11*1/10+3/10*2/9*9/11*8/10+(7/10*3/9+3/10*7/9)*(9/11*2/10+2/11*9/10)
p1=1-3/10*2/9*2/11*1/10

def comb(k, n):
 return factorial(n) / (factorial(k) * factorial(n - k))

#первая и вторая корзина по 2 белых мяча
p4 = comb(2, 7) / comb(2, 10) * comb(2, 9) / comb(2, 11)

#первая или вторая корзина 2 белых мяча
p2 = (comb(2, 7) / comb(2, 10) * comb(2, 2) / comb(2, 11)) + (comb(2, 3) / comb(2, 10) * comb(2, 9) / comb(2, 11))

#первая и вторая корзина по одному белому мячу
p2 += comb(1, 7) * comb(1, 3) /  comb(2, 10) * comb(1, 9) * comb(1, 2) / comb(2, 11)

#обратная вероятность вытащить ноль белых мячей
p1 = 1 - comb(0, 7) / comb(7, 10) * comb(0, 9) / comb(9, 11)


print(f'Ответ 2.4.1 - все мячи белые = {p4:.5f}')
print(f'Ответ 2.4.2 - 2 мяча белые = {p2:.5f}')
print(f'Ответ 2.4.3 - хотя бы 1 мяч белый = {p1:.5f}')

Ответ 2.4.1 - все мячи белые = 0.30545
Ответ 2.4.2 - 2 мяча белые = 0.20485
Ответ 2.4.3 - хотя бы 1 мяч белый = 0.99985


## Домашнее задание №3

**Задача 3.1.**

Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.

In [16]:
import numpy as np
from math import factorial
arr=np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])
arr

def mean_value(array):
    return sum(array)/len(array)

print(f'Среднее арифметическое М(Х) = {mean_value(arr): .2f}')

Среднее арифметическое М(Х) =  65.30


In [45]:
def mean_square_deviation(array):
    square_dev=(array-mean_value(array))**2
    return (sum(square_dev)/len(square_dev))**(1/2)
print(f'Среднее квадратичное отклонение D(X) = {mean_square_deviation(arr): .4f}')

Среднее квадратичное отклонение D(X) =  30.8239


In [41]:
def dispers(array, no_off=False):
#аргумент no_off отвечает за расчет смещенной, или несмещенной оценки дисперсии. По умолчанио расчитывается смещенная.#
    square_dev=(array-mean_value(array))**2
    return sum(square_dev)/(len(square_dev)-1) if no_off else sum(square_dev)/len(square_dev)

Среднее квадратичное отклонение для данной выборки SD =  30.8239
